DOC-818 K8s Galactica release changes (#1527)
RedisLabs/redislabs-docs: content/platforms/release-notes/k8s-6-2-4-tbd-2021-09.md

---
Title: Redis Enterprise for Kubernetes Release Notes 6.0.20-<tbd> (Sept 2021)
linktitle: 6.2.4-<tbd> (Sept 2021)
description: Release notes for version 6.2.4-<tbd>
weight: 67
alwaysopen: false
categories: ["Platforms"]
---
## Overview

> The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-55) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref ""/rs/release-notes/rs-6-2-4-august-2021/>}}) and includes several enhancements and bug fixes.

Author: ```suggestion
The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-55) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref "/rs/release-notes/rs-6-2-4-august-2021/">}}) and includes several enhancements and bug fixes.
```

> The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-55) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref "/rs/release-notes/rs-6-2-4-august-2021/">}}) and includes several enhancements and bug fixes.

Author: ```suggestion
The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-55) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref "/rs/release-notes/rs-6-2-4-august-2021.md">}}) and includes several enhancements and bug fixes.
```

The key new features, bug fixes, and known limitations are described below.

## Images

This release includes the following container images:

> * **Redis Enterprise**: `redislabs/redis:6.2.4-55` or  redislabs/redis:6.2.4-55.rhel7-openshift`

Author: ```suggestion
* **Redis Enterprise**: `redislabs/redis:6.2.4-55` or  `redislabs/redis:6.2.4-55.rhel7-openshift`
```

---
Title: Redis Enterprise for Kubernetes Release Notes 6.0.20-<tbd> (Sept 2021)
linktitle: 6.2.4-<tbd> (Sept 2021)
description: Release notes for version 6.2.4-<tbd>
weight: 67
alwaysopen: false
categories: ["Platforms"]
---
## Overview

> The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-55) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref "/rs/release-notes/rs-6-2-4-august-2021.md">}}) and includes several enhancements and bug fixes.

Reviewer: The K8s 6.2.4-55 link currently gives me a 404, but maybe this is just a GitHub permissions issue on my part though?

> The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-55) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref "/rs/release-notes/rs-6-2-4-august-2021.md">}}) and includes several enhancements and bug fixes.

Reviewer 2: ```suggestion
The Redis Enterprise [K8s 6.2.4-55](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.2.4-1) release provides support for the [Redis Enterprise Software release 6.2.4]({{<relref "/rs/release-notes/rs-6-2-4-august-2021.md">}}) and includes several enhancements and bug fixes.
```
@kaitlynmichael @rrelledge I need to tag the release. likely late on 9/14.

The key new features, bug fixes, and known limitations are described below.

## Images

This release includes the following container images:

* **Redis Enterprise**: `redislabs/redis:6.2.4-55` or  `redislabs/redis:6.2.4-55.rhel7-openshift`

> * **Operator**: `redislabs/operator:6.2.4-<tbd>`

Reviewer 2: ```suggestion
* **Operator**: `redislabs/operator:6.2.4-1`
```

> * **Services Rigger**: `redislabs/k8s-controller:6.0.20-<tbd>` or `redislabs/services-manager:6.2.4-<tbd>` (on the Red Hat registry)

Reviewer 2: ```suggestion
* **Services Rigger**: `redislabs/k8s-controller:6.0.20-1` or `redislabs/services-manager:6.2.4-1` (on the Red Hat registry)
```

## New features

* Internode encryption configuration through K8s custom resources (RED-59699, RED-60318)

## Feature improvements

> * Support for addition attribute in REDB secret containing comma seperated list of service names (RED-48469)

Reviewer 2: ```suggestion
* Support for addition attribute in REDB secret containing comma separated list of service names (RED-48469)
```

* Support OpenShift 4.8 (K8s 1.21) (RED-59424)
* Support K8s 1.21 - GKE (RED-59048)
* Support K8s 1.21 - kOps (RED-59047)
* Support K8s 1.19-1.21 - EKS (RED-60287)
* Support K8s 1.19, 1.20 - AKS (RED-59050)
* Support K8s 1.20 - Rancher (RED-59049)

## Fixed bugs

> * Fixed issue with RS pods not recovering from container fails (RED-)

Reviewer: Missing RED- ticket number?

> * Fixed issue with RS pods not recovering from container fails (RED-)

Reviewer 2: ```suggestion
* Fixed issue with RS pods not recovering from container failures (RED-53042)
```

> * Fixed rare problem of RS pod restarting too early while statefulSet was rolling out, causing quorum loss (RED-53042)

Reviewer 2: ```suggestion
* Fixed rare problem of Redis Enterprise pod restarting too early while statefulSet was rolling out, causing quorum loss (RED-53042)
```

### Readiness probe incorrect on failures (RED-39300)

STS Readiness probe does not mark a node as "not ready" when running `rladmin status` on the node fails.

### Role missing on replica sets (RED-39002)

The `redis-enterprise-operator` role is missing permission on replica sets.

### Private registries are not supported on OpenShift 3.11 (RED-38579)

Openshift 3.11 does not support DockerHub private registries. This is a known OpenShift issue.

### Internal DNS and Kubernetes DNS may have conflicts (RED-37462)

DNS conflicts are possible between the cluster mdns_server and the K8s DNS. This only impacts DNS resolution from within cluster nodes for Kubernetes DNS names.

### 5.4.10 negatively impacts 5.4.6 (RED-37233)

Kubernetes-based 5.4.10 deployments seem to negatively impact existing 5.4.6 deployments that share a Kubernetes cluster.

### Node CPU usage is reported instead of pod CPU usage (RED-36884)

In Kubernetes, the node CPU usage we report on is the usage of the Kubernetes worker node hosting the REC pod.

### Clusters must be named "rec" in OLM-based deployments (RED-39825)

> In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) is bound to a specific service account name (i.e., "rec"). The workaround is to name the cluster "rec".

Reviewer: ```suggestion
In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) are bound to a specific service account name (i.e., "rec"). The workaround is to name the cluster "rec".
```

### REC clusters fail to start on Kubernetes clusters with unsynchronized clocks (RED-47254)

When REC clusters are deployed on Kubernetes clusters with unsynchronized clocks, the REC cluster does not start correctly. The fix is to use NTP to synchronize the underlying K8s nodes.

### Deleting an OpenShift project with an REC deployed may hang (RED-47192)

When a REC cluster is deployed in a project (namespace) and has REDB resources, the
REDB resources must be deleted first before the REC can be deleted. As such, until the
REDB resources are deleted, the project deletion will hang. The fix is to delete the
REDB resources first and the REC second. Afterwards, you may delete the project.

### REC extraLabels are not applied to PVCs on K8s versions 1.15 or older (RED-51921)

In K8s 1.15 or older, the PVC labels come from the match selectors and not the
PVC templates. As such, these versions can not support PVC labels. If this feature
is required, the only fix is to upgrade the K8s cluster to a newer version.

### REC might report error states on initial startup (RED-61707)

> There is not workaround at this time except to ignore the errors.

Reviewer: ```suggestion
There is no workaround at this time except to ignore the errors.
```

### PVC size issues when using decimal value in spec (RED-62132)

The workaround for this issue is to make sure you use integer values for the PVC size.

## Compatibility Notes
See [Supported Kubernetes distributions]({{<relref "/platforms/kubernetes/reference/supported_k8s_distributions.md">}}) for the full list of supported distributions.

### Now supported

* OpenShift 4.8
* GKE K8s version 1.21
* kOps K8s version 1.21
* EKS K8s versions 1.19-1.21
* AKS K8s versions 1.19-1.20
* Rancher K8s version 1.20

### No longer supported

* GKE K8s version 1.17 (previously deprecated) 
* kOps K8s version 1.15 (previously deprecated) 

## Deprecation notice

* kOps 1.16 and 1.17 are deprecated

> * VMWare TKGIE 1.7 (K8s 1.16), VMWare TKGIE 1.8 (K8s 1.17) are deprecated (no longer supported by VMWare)

Reviewer 2: @kaitlynmichael additional deprecation
```suggestion
* VMWare TKGIE 1.7 (K8s 1.16), VMWare TKGIE 1.8 (K8s 1.17) are deprecated (no longer supported by VMWare)
* Openshift 3.11 (K8s 1.11) is now deprecated. Redis will continue to support existing deployments for the lifetime of Openshift 3.11, but new deployments are strongly discouraged.
```
